lib/util/db.js

```diff
@@ -18,6 +18,45 @@ const { PartialPipe, mapStream } = require('../util/stream');
 const { construct } = require('../util/util');
 
 
+////////////////////////////////////////////////////////////////////////////////
+// DATABASE CONFIG
+
+const validateConfig = (config) => {
+  const { host, port, database, user, password, ssl, ...unsupported } = config;
+
+  if (!(ssl == null || ssl === true || ssl === false))
+    throw Problem.internal.invalidDatabaseConfig({ reason: 'ssl can only be true or false.' });
+
+  const unsupportedKeys = Object.keys(unsupported);
+  if (unsupportedKeys.length !== 0)
+    throw Problem.internal.invalidDatabaseConfig({
+      reason: `'${unsupportedKeys[0]}' is not supported.`
+    });
+};
+
+// Returns a connection string that will be passed to Slonik.
+const connectionString = (config) => {
+  validateConfig(config);
```

Comment:
Return a Problem or null? Or an Option of a Problem? If there is a Problem here, throw it.